ZKUI-152: Handle token expired error

JBWatenbergScality opened the pull request.

JBWatenbergScality committed:
ZKUI-152: Handle token expired error

bert-e commented:
# Hello jbwatenbergscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko-ui/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZKUI-152 contains:


* `1.5.0`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `1.4.6`

* `1.5.0`


Please check the `Fix Version/s` of ZKUI-152, or the target
branch of this pull request.

JBWatenbergScality committed:
ZKUI-152: Handle token expired xml error.

GetRolesForWebIdentity endpoint classical response is returned using json.
However, with the same accept: application/json header, in case of error the returned error is (or may be) formated using XML.
This is the case for the TokenExpired error response which is returned with 400 status code.
As the UI should display a re-login dialog when receiving this specific 400 response, we need to parse it.
Hence this try and fall response parsing mechanism that is super ugly :see_no_evil:

ChengYanJin reviewed (approved):
LGTM! Let's go since this is the only way to handle this token expired error.

JBWatenbergScality commented:
/approve

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZKUI-152 contains:


* `1.5.0`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `1.4.6`

* `1.5.0`


Please check the `Fix Version/s` of ZKUI-152, or the target
branch of this pull request.




*The following options are set:* **approve**

JBWatenbergScality commented:
/status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/ZKUI-152-handle-token-expired-error` into
`development/1.4`
* [w/1.5/feature/ZKUI-152-handle-token-expired-error](https://github.com/scality/zenko-ui/compare/development/1.5...w/1.5/feature/ZKUI-152-handle-token-expired-error) will be merged into `development/1.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.4`

* :heavy_check_mark: `development/1.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.4`

* :heavy_check_mark: `development/1.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue ZKUI-152.

Goodbye jbwatenbergscality.
